Enterprise-CMCS/cmcs-eregulations
# EREGCSC-1765 -- Vue 3 Upgrade

Resolves [EREGCSC-1765](https://jiraent.cms.gov/browse/EREGCSC-1765)

**Description**

Upgrading Vue 2 to Vue 3.

**This pull request changes:**

- Upgraded Vue to 3.4.21 (latest)
- Upgraded Vuetify to 3.5.11 (latest minor version)
- Upgraded Vue Router to 4.3 (latest)
- Changed/refactored/updated everything that needed to be addressed to get the site building and working as expected

**Steps to manually verify this change:**

1. Green check marks
   - If so desired, you can also check the Vitest unit tests by running `npm run test-vue` at `cmcs-eregulations/solution/ui/regulations`.  There are 81 tests in 18 test files and all should pass.
2. Visit [experimental deployment](https://fvckxcjhbk.execute-api.us-east-1.amazonaws.com/dev1218) and make sure site works as expected with no strangeness in the console
3. Attempt to build locally if so desired
